docs/userguide/core-workflow.md

```diff
@@ -97,73 +108,180 @@ struct State {
+    let sink = context.makeSink(of: Action.self)
+
+    return DemoScreen(
+        title: "A nice title",
+        onTap: { sink.send(Action.refreshButtonTapped) }
+}
+TK
+```
+
+```Kotlin tab=
+TK
+```
+
+## Workflows form a hierarchy (they may have children)
 
 As they produce a `Rendering` value, it is common for workflows to delegate some portion of that
-work to a _child workflow_. This is also done via the `RenderContext` that is passed into the
-`render` method. In order to delegate to a child, the parent workflow instantiates the child within
-the `render` method. The parent then calls `render` on the context, with the child workflow as the
-single argument. The infrastructure will spin up the child workflow (including initializing its
-initial state) if this is the first time this child has been used, or, if the child was also used on
-the previous `render` pass, the existing child will be updated. Either way, `render` will ultimately
-be called on the child (by the Workflow infrastructure), and the resulting `Child.Rendering` value
-will be returned to the parent.
+work to a _child workflow_. This is done via the `RenderContext` that is passed into the `render`
+method. In order to delegate to a child, the parent calls `renderChild` on the context, with the
+child workflow as the single argument. The infrastructure will spin up the child workflow (including
+initializing its initial state) if this is the first time this child has been used, or, if the child
+was also used on the previous `render` pass, the existing child will be updated. Either way,
+`render` will immediately be called on the child (by the Workflow infrastructure), and the resulting
+child's `Rendering` value will be returned to the parent.
 
 This allows a parent to return complex `Rendering` types (such as a view model representing the
 entire UI state of an application) without needing to model all of that complexity within a single
 workflow.
 
-### Workflows can respond to UI events
-
-The `RenderContext` that is passed into `render` as the second parameter provides some useful tools
-to assist in creating the `Rendering` value.
+!!! info "Workflow Identity"
+    The Workflow infrastructure automatically detects the first time and the last subsequent time
+    you've asked to render a child workflow, and will automatically initialize the child and clean
+    it up. In both Swift and Kotlin, this is done using the workflow's concrete type. Both languages
+    use reflection to do this comparison (e.g. in Kotlin, the workflows' `KClass`es are compared).
 
-If a workflow is producing a view model, it is common to need an event handler to respond to UI
-events. The `RenderContext` has API to create an event handler that, when called, will advance the
-workflow by dispatching an action back to the workflow.
+    It is an error to render workflows of the same type more than once in the same render pass.
```

Comment:
Since we give you a way to render the same type, it feels strange to say it’s an error.

Maybe something like “It is an error to render the same child workflow more than once in the same. Since type is used for workflow identity, the child rendering APIs take an optional string key to differentiate between multiple child workflows of the same type.”